src/OpenIddict.Mvc/OpenIddictExtensions.cs

```diff
@@ -154,6 +154,26 @@ public static class OpenIddictExtensions {
                     return provider.GetRequiredService(typeof(OpenIddictUserManager<>).MakeGenericType(builder.UserType));
                 });
 
+                // Register the OpenIddict services in the isolated container.
+                services.AddScoped(typeof(OpenIddictServices<,,,,>).MakeGenericType(
```

Comment:
nit: we are making a slow reflection call `typeof(OpenIddictServices<,,,,>).MakeGenericType(....)` twice, this type can be saved and reused in the `container.GetRequiredService(typeof(OpenIddictServices<,,,,>).MakeGenericType(...)`